src/lib/core/compatibility/compatibility.ts

```diff
@@ -171,7 +171,7 @@ export class CompatibilityModule {
   }
 
   constructor(@Optional() @Inject(DOCUMENT) document: any) {
-    if (isDevMode() && typeof document && !document.doctype) {
+    if (isDevMode() && document && !document.doctype) {
```

Comment:
It won't, because it'll hit the parameter, before it tries to look for it globally.